File: Data Structures & Algorithms/duplicate-integer/submission-1.rs
use std::collections::HashSet;

impl Solution {
    pub fn has_duplicate(nums: Vec<i32>) -> bool {
        let mut seen = HashSet::new();
        
        for num in nums {
            // insert() returns `false` if the item was already in the set
            if !seen.insert(num) {
                return true; // We found a duplicate!
            }
        }
        
        false // If we finish the loop, everything was unique
    }
}

File: Data Structures & Algorithms/find-duplicate-integer/submission-1.cpp
#include <cmath>

class Solution {
public:
    int findDuplicate(vector<int>& nums) {
     for(int num: nums){
        nums[std::abs(num)] = -1 * nums[std::abs(num)];
        if(nums[std::abs(num)] > 0) {return std::abs(num);}
     }
     return -1;
    }
    
};
